src/nora/components/NoraTextAreaInput/NoraTextAreaInput.js

```diff
@@ -17,7 +18,76 @@ export const NoraTextAreaInput = ({
   onChange,
   ...rest
 }) => {
+  const {
+    formChangeHandler,
+    validator,
+    initialValue,
+    currentValue,
+    currentError,
+    setFieldTouched,
+  } = { ...rest }
+
   const classes = [styles.NoraTextInput, textClassName].join(' ')
+
+  // Set up validation hooks
+  const [getError, setError, , validate] = useErrorMessage(validator)
+
+  const [val, setVal] = useState(currentValue || initialValue || '')
+
+  const [touched, setTouched] = useState(initialValue ? true : false)
+
+  const [doValidation] = useInputValidation({
+    validate,
+    setError,
+    formChangeHandler,
+  })
+
+  const handleChange = (ev) => {
+    const val = ev.target.value
+    setVal(val)
+
+    // We call setTouched in onBlur, so can reliably call getter here
+    doValidation(val, touched)
+
+    if (onChange) onChange(ev)
+  }
+
+  const setAllTouched = () => {
+    // We set touched to change the react state, but it's async and
+    // processing still, so, we use a flag for doValidation
+    setTouched(true)
+    // Also tell the form we've been touched
+    if (setFieldTouched) {
+      setFieldTouched(true)
+    }
+  }
+
+  // Initial Value aka prefilled—are considered "touched", but must prevalidate
+  // which will in turn update the internal form state as to their validity
+  useEffect(() => {
+    if (!!formChangeHandler && initialValue) {
```

Comment:
Similar to above, why is this logic using `initialValue` and not `value` or `resolvedValue`?

In the checkbox, we had to be backwards compatible. But shouldn't we be able to use only `resolvedValue` here?

Reply:
Updated